### ISSUE TITLE
Non-deterministic outputs in eval for FT Transformer (related to hyperconnections?)

### ISSUE THREAD

**Issue by 0xandrewdu** [lucidrains/tab-transformer-pytorch#28]
I trained a FT transformer with a deterministic dataloader / data processing pipeline, and during the training the validation loss looked good. However, on loading the weights of a checkpoint back into the same model (with identical params), the validation loss is suddenly completely different than when the model was originally evaluated. Furthermore, the during-training val loss is always around some value, while the validation loss is varied but consistently worse than the original run. I have verified that the model weights and data are all the same across eval runs, but I have found that the model predictions vary wildly given an identical batch. At the same time, calling `model(torch.ones_like(batch[0].unsqueeze(-1)))` gives the same value every time. It is unclear to me what this source of performance loss is, as I have plugged a basic MLP into the same pipeline and cannot reproduce this randomness. Is there potentially something with hyperconnections that breaks down on model saving/loading?

**Comment by lucidrains** [comment]
hmm, i don't know, you can test this right? just turn off hyper connections and do a short run

**Comment by lucidrains** [comment]
@0xandrewdu yea no problem Andrew

could you share whether you see a big improvement with hyperconnections?

**Comment by 0xandrewdu** [comment]
I have not tried using this without hyperconnections yet—is there a quick way to disable it?

**Comment by 0xandrewdu** [comment]
I just finished a quick test of four versus one residual stream; qualitatively, the model with 4 does converge appreciably faster in its loss, though the final losses of the two models aren't all that different in terms of where they plateau. Thank you for your work in implementing this!

**Comment by lucidrains** [comment]
@0xandrewdu thank you Andrew! 🙏
